components/SpotDetails/SpotSocial/SpotSocial.js

```diff
@@ -1,24 +1,57 @@
 import SpotSocialComment from "./SpotSocialComment";
 import SpotSocialForm from "./SpotSocialForm";
-import { useState } from "react";
 import styled from "styled-components";
 
-export default function SpotSocial({ ID }) {
-  const [comments, setComments] = useState([]);
-
+export default function SpotSocial({ SpotData }) {
   function handleNewComment(comment) {
-    setComments([...comments, comment]);
+    let dbComments = SpotData.comments.map((comment) => {
+      return { text: comment.text, _id: comment._id };
+    });
+    dbComments.push(comment);
+
+    const updatedSpot = {
+      _id: SpotData._id,
+      slug: SpotData.slug,
+      name: SpotData.name,
+      image: SpotData.image,
+      country: SpotData.country,
+      city: SpotData.city,
+      zip: SpotData.zip,
+      street: SpotData.street,
+      coordinates: SpotData.coordinates,
+      latitude: SpotData.latitude,
+      longitude: SpotData.longitude,
+      description: SpotData.description,
+      winddirection: SpotData.winddirection,
+      surfcenter: SpotData.surfcenter,
+      parking: SpotData.parking,
+      camping: SpotData.camping,
+      comments: dbComments,
+    };
+
+    useChange(updatedSpot);
+    window.location.reload(false);
+  }
+
+  async function useChange(data) {
+    await fetch("/api/" + data._id, {
```

Comment:
The prefix `use` should be reserved for React hooks (and this isn't a hook). Please use a different name (e.g. `uploadChanges`)